### PR TITLE
hotplug_mem: Add case for memory-backend-memfd

### PR DESCRIPTION
**Pull request by huangyum**
Signed-off-by: Yumei Huang <yuhuang@redhat.com>
ID: 1932729
Depends on https://github.com/avocado-framework/avocado-vt/pull/2932.

### CODE REVIEW

**Comment by huangyum** [comment]
(1/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.hotplug_memory.after.guest_reboot.unplug.pluged_memory.backend_memfd.policy_default.two.q35: PASS (93.09 s)
 (2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.hotplug_memory.after.guest_reboot.hotplug.backend_memfd.policy_default.two.q35: PASS (151.58 s)

@yanan-fu @MiriamDeng  Pls help review, thanks.

**Comment by yanan-fu** [comment]
Suggest add `required_qemu = [4.2.0,)` , is it ?

**Comment by huangyum** [comment]
> Suggest add `required_qemu = [4.2.0,)` , is it ?

I added host restriction, it should restrict qemu version as well.

File: qemu/tests/cfg/hotplug_mem.cfg [hunk 1]
```diff
@@ -113,6 +113,14 @@
             mem-path = /mnt/kvm_hugepage
             pre_command = "echo 3 > /proc/sys/vm/drop_caches"
             pre_command_noncritical = yes
+        - backend_memfd:
+            no Host_RHEL.m7 Host_RHEL.m8.u0 Host_RHEL.m8.u1 Host_RHEL.m8.u2
+            backend_mem = memory-backend-memfd
+            setup_hugepages = yes
+            hugetlb_mem = yes
+            only hotplug unplug.pluged_memory
```
Comment (MiriamDeng) [inline qemu/tests/cfg/hotplug_mem.cfg:121]: For rhel7 guest, the unplug is not supported.

Comment (huangyum) [reply qemu/tests/cfg/hotplug_mem.cfg:121]: Thanks, add it for all unplug cases.

**Comment by huangyum** [comment]
@MiriamDeng @yanan-fu Do you have more comments?

**Comment by yanan-fu** [comment]
> > Suggest add `required_qemu = [4.2.0,)` , is it ?
> 
> I added host restriction, it should restrict qemu version as well.

I am not sure if it have restriction to kernel or other pkg version, but i remember RHEL8.2 advance virt use qemu 4.2 too, do we need to cover it ?

**Comment by huangyum** [comment]
> > > Suggest add `required_qemu = [4.2.0,)` , is it ?
> > 
> > 
> > I added host restriction, it should restrict qemu version as well.
> 
> I am not sure if it have restriction to kernel or other pkg version, but i remember RHEL8.2 advance virt use qemu 4.2 too, do we need to cover it ?

I don't think so. It's enabled since 8.2.1.

**Comment by yanan-fu** [comment]
> > > > Suggest add `required_qemu = [4.2.0,)` , is it ?
> > > 
> > > 
> > > I added host restriction, it should restrict qemu version as well.
> > 
> > 
> > I am not sure if it have restriction to kernel or other pkg version, but i remember RHEL8.2 advance virt use qemu 4.2 too, do we need to cover it ?
> 
> I don't think so. It's enabled since 8.2.1.

Could you explain this with more details ?  
From the manual case, it show this feature was enabled since qemu 4.2,   So it is enabled since 8.2.1 or since qemu 4.2 ?  They are different because RHEL8.2.0 already start use qemu 4.2, thanks!

**Comment by huangyum** [comment]
> > > > > Suggest add `required_qemu = [4.2.0,)` , is it ?
> > > > 
> > > > 
> > > > I added host restriction, it should restrict qemu version as well.
> > > 
> > > 
> > > I am not sure if it have restriction to kernel or other pkg version, but i remember RHEL8.2 advance virt use qemu 4.2 too, do we need to cover it ?
> > 
> > 
> > I don't think so. It's enabled since 8.2.1.
> 
> Could you explain this with more details ?
> From the manual case, it show this feature was enabled since qemu 4.2, So it is enabled since 8.2.1 or since qemu 4.2 ? They are different because RHEL8.2.0(advance virt) already start use qemu 4.2, thanks!

It's enabled since 8.2.1. Not all qemu4.2 support it.

**Comment by yanan-fu** [comment]
> 
> It's enabled since 8.2.1. Not all qemu4.2 support it.

Thanks for your info,  `required_qemu` support the format such as `[4.2.0-23, )`, so it is not a blocker ^^
But, as we discussed offllline, it is make sense for to me start test from 8.3 since it was not enabled by upper layer before that.   So, Ack 

By the way, please remember to update the manual case as well ~

**Comment by yanan-fu** [comment]
@MiriamDeng  Do you have any more comment ? Thanks!

**Comment by MiriamDeng** [comment]
ACK, thanks.
 (1/2) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.hotplug_memory.after.guest_reboot.unplug.pluged_memory.backend_memfd.policy_default.two: PASS (310.00 s)
 (2/2) Host_RHEL.m8.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.ppc64le.io-github-autotest-qemu.hotplug_memory.after.guest_reboot.hotplug.backend_memfd.policy_default.two: PASS (240.35 s)

**Pull request merged**